protobuf_serialization/libp2p_varint.nim

```diff
@@ -0,0 +1,392 @@
+## Nim-Libp2p
+## Copyright (c) 2018 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+## This module implements Variable Integer `VARINT`.
+## This module supports two variants of variable integer
+## - Google ProtoBuf varint, which is able to encode full uint64 number and
+##   maximum size of encoded value is 10 octets (bytes).
+##   https://developers.google.com/protocol-buffers/docs/encoding#varints
+## - LibP2P varint, which is able to encode only 63bits of uint64 number and
+##   maximum size of encoded value is 9 octets (bytes).
+##   https://github.com/multiformats/unsigned-varint
+import bitops, typetraits
```

Comment:
this needs to be ported to bitops2 - `bitops` has some UB that we don't really want to be burdened by